soccer/src/soccer/soccermom/soccermom.cpp

```diff
@@ -0,0 +1,58 @@
+#include <chrono>
+#include <functional>
+#include <memory>
+#include <string>
+
+#include <rclcpp/rclcpp.hpp>
+#include "std_msgs/msg/string.hpp"
+
+#include <rj_utils/logging.hpp>
+#include "global_params.hpp"
+#include "soccermom.hpp"
+
+namespace tutorial {
+
+SoccerMom::SoccerMom()
+    : rclcpp::Node{"tutorial", rclcpp::NodeOptions{}
+                        .automatically_declare_parameters_from_overrides(true)
+                        .allow_undeclared_parameters(true)} { // if the subscriber doesn't work, rj_msgs/msg/TeamColor as the topic  // /referee/team_color
+    //subscription = this->create_subscription<std_msgs::msg::String>("topic", 10, std::bind(&SoccerMom::topic_callback, this, std::placeholders::_1));
+    subscription = create_subscription<rj_msgs::msg::TeamColor>(
+        referee::topics::kTeamColorTopic, rclcpp::QoS(1).transient_local(),
+        [this](rj_msgs::msg::TeamColor::SharedPtr color) {  // NOLINT
+            if (color->is_blue) {
+                teamColor = "blueberries";
+            } else {
+                teamColor = "banana";
+            }
+            //RCLCPP_INFO(this->get_logger(), teamColor);  
+        });
+    publisher = this->create_publisher<std_msgs::msg::String>("/team_fruit", 10);
+    timer = this->create_wall_timer(std::chrono::milliseconds(500), std::bind(&SoccerMom::timer_callback, this));
```

Comment:
using a wall timer isn't necessarily incorrect, but we can be more precise in this case. since we know that the fruit is only updated with team color, we ought to only publish in reaction to a new team color (in other words, directly in your subscription callback)

Reply:
this prevents both over and under-publishing